mama/c_cpp/src/c/datetime.c

```diff
@@ -988,14 +1049,14 @@ mama_status mamaDateTime_getAsString (const mamaDateTime dateTime,
     if (!dateTime || !buf)
         return MAMA_STATUS_INVALID_ARG;
 
-    seconds = (time_t) mamaDateTimeImpl_getSeconds(*dateTime);
+    seconds = (time_t) mamaDateTimeImpl_getSeconds((mama_time_t*)dateTime);
```

Comment:
There's an awful lot of casting of dateTime in this method. I would suggest a mama_date_time_t* assigned at the top to make this less sore on the eyes. Overkill in some of the other methods but it's used a lot in this one.